### PR TITLE
Remote Config Update Function

### PR DESCRIPTION
**Pull request by stdavis**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
   "scripts": {
     "build": "vite build",
     "dev": "cross-env NODE_ENV=development run-p dev:*",
-    "dev:firebase": "firebase emulators:start --import .emulator-data --only functions",
+    "dev:firebase": "export GOOGLE_APPLICATION_CREDENTIALS=\"$PWD/src/functions/functions-key-dev.json\" && firebase emulators:start --import .emulator-data --only functions",
```
Comment (steveoh) [inline package.json:13]: **suggestion**: use https://www.npmjs.com/package/cross-env to be more flexible

File: src/functions/update.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    console.log('no changes detected, skipping remote config update');
+    return;
+  }
+
+  template.parameters.queryLayers.defaultValue.value = queryLayers;
+  template.parameters.relatedTables.defaultValue.value = relatedTables;
+  template.parameters.links.defaultValue.value = links;
+
+  console.log('validating new template');
+  await remoteConfig.validateTemplate(template);
```
Comment (steveoh) [inline src/functions/update.js:115]: **question**: does this throw if it's not valid?

**Review by steveoh — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (steveoh) [reply package.json:13]: https://superuser.com/questions/295021/windows-powershell-equivalent-to-unix-linux-pwd

¯\_(ツ)_/¯

File: src/functions/update.js (hunk 2 above)
Comment (stdavis) [reply src/functions/update.js:115]: Yes and provides a nice error message about what's going on.

**Pull request merged**